API/testrunner/devices/device_base.py

```diff
@@ -84,3 +84,32 @@ def logout(self):
         Logout from the device.
         '''
         self.channel.close()
+
+    def iotjs_build_info(self):
+        '''
+        Get buildinfo from iotjs
+        '''
+        if self.device in ['rpi2', 'artik530']:
+            iotjs = '%s/iotjs' % self.workdir
+            buildinfo = '%s/tests/tools/iotjs_build_info.js' % self.workdir
+            command = ' '.join([iotjs, buildinfo])
+
+            if self.device == 'rpi2':
+                self.login()
+
+        elif self.device in ['artik053', 'stm32f4dis']:
+            command = 'iotjs /test/tools/iotjs_build_info.js'
+            self.reset()
+            self.login()
```

Comment:
I think the `reset` is not necessary in this case. (That is necessary if the js file has async operations.)

There are too many `if` statements. I think this code can be simplified (pseudo code):

```
if device is rpi2 or artik530:
  command = ...
elif device is stm32 or artik053:
  command = ...

login (for every device)
run the command

builtins = ...
features = ...
stability = ...

logout (for every device)

return ...
```